libethereum/BlockQueue.h

```diff
@@ -67,6 +67,117 @@ enum class QueueStatus
 
 std::ostream& operator<< (std::ostream& os, QueueStatus const& obj);
 
+template<class T>
+class SizedBlockQueue
+{
+public:
+	std::size_t count() const
+	{
+		return m_queue.size();
+	}
+
+	std::size_t size() const
+	{
+		return m_size;
+	}
+	
+	bool isEmpty() const
+	{
+		return m_queue.empty();
+	}
+
+	h256 nextHash() const
+	{
+		return m_queue.front().verified.info.sha3Uncles();
+	}
+
+	T const& next() const
+	{
+		return m_queue.front();
+	}
+
+	template<class Op>
+	void forEach(Op _op) const
+	{
+		std::for_each(m_queue.cbegin(), m_queue.cend(), _op);
+	}
+
+	void clear()
+	{
+		m_queue.clear();
+		m_size = 0;
+	}
+
+	void enqueue(T&& _t)
+	{
+		m_size += _t.blockData.size();
```

Comment:
Although, yeah, it would be nicer to emplace_back first, then update m_size, I'll change it